Response model

HunkevychPhilip opened the pull request.

hardworking-gopher committed:
Update composer.json

hardworking-gopher committed:
Dictionary class, entry builders

hardworking-gopher committed:
Update MainController.php

hardworking-gopher committed:
Demonstrating skeleton

hardworking-gopher committed:
Delete frameworkDescriptionVersion1.1.4.xsd

hardworking-gopher committed:
Composer.json

hardworking-gopher committed:
Exceptions handling and rerouting

hardworking-gopher committed:
Templates update

yderevianko-cartera commented on src/Controller/SearchController.php:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * @Route("/search", name="app_search", methods="GET")
+     */
+    public function index(): Response
```
why did you named controller action index if it has route /search ?

HunkevychPhilip replied:
Let me get this straight: are controller methods' identifiers must match route names they related to? For instance, this "index" method must be named "search" instead?

yderevianko-cartera replied:
it should be similar to url
for example:
/book/new => new
/book/{id}/edit => edit
/search => search

yderevianko-cartera commented on src/Controller/SearchController.php:
```diff
@@ -0,0 +1,56 @@
+        $lang = $request->query->get('language');
+
+        if (empty($word)) {
+            header("Location: /?error=you_must_enter_the_word");
```
please do not use die and header (not best practice)

You can render error template if you are missing them

HunkevychPhilip replied:
Of course, I've done it this way so you can make the issue about how it should be instead. I don't really know how to handle exceptions right now. Thank you, I'll make error templates.

yderevianko-cartera commented on src/Controller/SearchController.php:
```diff
@@ -0,0 +1,56 @@
+            $results = $dictionary->entries($lang, $word);
+        } catch (DictionaryException $e) {
+            $error = $e->getMessage();
+            header("Location: /?error=$error");
```
same as above, try to render error template or same template but with error messages

yderevianko-cartera commented on src/System/Client/ClientException.php:
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\System\Client;
```
do not name folder System try using Service

yderevianko-cartera commented on src/System/v2/Builders/EntriesBuilder.php:
```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\System\v2\Builders;
```
why v2 ? and why do you need this Builder ?

HunkevychPhilip replied:
I thought "v2" is the proper way of calling these folders because they use base uri which ends like "/v2", so that can be some version name etc. I'll remove it.

The unneeded namespace is my fault.

yderevianko-cartera commented on src/System/v2/Dictionary.php:
```diff
@@ -0,0 +1,50 @@
+        } catch (ClientException $e) {
+            if ($e->getCode() === 404) {
+                header("Location: /?error=404:page_not_found");
+                die();
```
you application should never die
please throw Exception instead

yderevianko-cartera commented on templates/main/homepage.html.twig:
```diff
@@ -1,16 +1,29 @@
-                    <label for="word" class="cover-heading">Enter your word:</label>
-                    <input type="text" name="word" minlength="1" maxlength="30" pattern="[\w]*" required>
+
+                    <section>
```
If you have time read about Symfony Forms

hardworking-gopher committed:
Configs

hardworking-gopher committed:
Refactor project structure

Namespaces, folders were refactored. Symfony form was added. 'ResponseHandler' logic was impoved.

hardworking-gopher committed:
Twig templates refactoring

hardworking-gopher committed:
Composer install error fix

hardworking-gopher committed:
Exceptions were created

hardworking-gopher committed:
Examples retrieving cycle

hardworking-gopher committed:
getExamples fix

hardworking-gopher committed:
regex

A regular expression to get a string contains only letters
